packages/vector_graphics_compiler/lib/src/paint.dart

```diff
@@ -240,95 +238,42 @@ enum GradientUnitMode {
 /// circle and [focalRadius] being the radius of that circle. If [focal] is
 /// provided and not equal to [center], at least one of the two offsets must
 /// not be equal to [Point.zero].
-@immutable
-class RadialGradient extends Shader {
+class RadialGradient extends Gradient {
   /// Creates a new radial gradient object with the specified properties.
   ///
   /// See [RadialGradient].
   const RadialGradient({
     required this.center,
     required this.radius,
-    required this.colors,
-    required this.offsets,
-    required this.tileMode,
-    this.transform,
+    required List<Color> colors,
+    required List<double> offsets,
+    required TileMode tileMode,
+    AffineMatrix? transform,
     this.focalPoint,
-    this.unitMode = GradientUnitMode.objectBoundingBox,
-  });
+    GradientUnitMode unitMode = GradientUnitMode.objectBoundingBox,
+  }) : super._(colors, offsets, tileMode, unitMode, transform);
 
   /// The central point of the gradient.
   final Point center;
 
   /// The colors to blend from the start to end points.
   final double radius;
 
-  /// The colors to gradate through.
-  final List<Color> colors;
-
-  /// The positions to apply [colors] to. If specified, must be the same length
-  /// as [colors]. If not specified, [colors] must be two colors.
-  final List<double> offsets;
-
-  /// Specifies the meaning of [from] and [to].
-  final TileMode tileMode;
-
-  /// The transform, if any, to apply to the gradient.
-  final AffineMatrix? transform;
-
   /// If specified, creates a two-point conical gradient using [center] and the
   /// [focalPoint].
   final Point? focalPoint;
 
-  /// Whether the coordinates in this gradient should be transformed by the
-  /// space this object occupies or or not.
-  final GradientUnitMode unitMode;
-
   @override
   RadialGradient applyBounds(Rect bounds, AffineMatrix transform) {
-    if (unitMode == GradientUnitMode.userSpaceOnUse) {
-      return RadialGradient(
-        center: transform.transformPoint(center),
-        radius: radius,
-        colors: colors,
-        offsets: offsets,
-        tileMode: tileMode,
-        transform: this.transform,
-        focalPoint: focalPoint == null
-            ? focalPoint
-            : transform.transformPoint(focalPoint!),
-        unitMode: unitMode,
-      );
-    }
-
     return RadialGradient(
```

Comment:
Don't we need to offset the gradient points conditional on the unitMode here?

Reply:
No, AFAIU it's because we're supplying the transform. We needed to do that before because we were artificially transposing them in the parsing method, which isn't happening now.

I suspect there's still some problem here, but I haven't been able to find an example that this breaks - though perhaps as we dig into #31 we'll find some.

Reply:
IOW: we only need to get the coordinates into the transformed space here, and we can let the transform take care of the rest, and we're basically always treating it as user space on use this way.

Reply:
Ahh I see, instead of transforming the coordinates just transform ... the transform. Makes sense!